src/use-swr.ts

```diff
@@ -92,21 +92,42 @@ const broadcastState: broadcastStateInterface = (key, data, error) => {
   }
 }
 
-const mutate: mutateInterface = (_key, data, shouldRevalidate = true) => {
+const mutate: mutateInterface = async (_key, _data, shouldRevalidate) => {
   const [key] = getKeyArgs(_key)
   if (!key) return
 
   // update timestamp
   MUTATION_TS[key] = Date.now() - 1
 
-  // update cached data
-  cacheSet(key, data)
+  let data, error
+
+  if (_data && typeof _data.then === 'function') {
```

Comment:
You *can* await a non-promise too. Not sure if there is a performance issue though
```suggestion
  const data = await _data;
```

Reply:
It will defer the execution order (await non-promise values), similar to `process.nextTick`. But the major reason is that for promises (which should be mutation requests), we don't have to revalidate again after the request in most cases.